### ISSUE TITLE
The latest versions of RetroArch do not create MAME playlists

### ISSUE THREAD

**Issue by xavierocampos** [libretro/RetroArch#10441]
This is not a post asking for help to generate a playlist because this can be done using an old version of Retroarch like version 1.7.5. This post is intended to make known a bug that has already been complained about by several people in the community and that for some reason is still dragging the latest versions of RetroArch: No MAME playlists are generated, for whatever reason.

I hope you can review it and fix it as I consider this a major bug, especially by order purists like me for whom manual scanning is not very helpful.

Greetings and thanks for your attention.

**Comment by Sanaki** [comment]
Enable "Scan without core match" in Playlists settings and they'll scan. That said, manual scan is still the best route, so long as you feed it a dat.

**Comment by marcecj** [comment]
I see this bug, too, in the Flatpak version.  Enabling "Scan without core match" as mentioned above helps, but shouldn't be necessary, as I have appropriate MAME cores installed already.  I do not recall seeing this behavior with other systems.

**Comment by markwkidd** [comment]
I have successfully scanned MAME playlists on more recent builds. Is this still an issue or can it be closed?

**Comment by Sanaki** [comment]
Given that arcade is based on a name match (and potentially stored hash matches), it'd require a completely new method added to the scanner to work properly. It's not going to magically resolve itself, because it's not a bug in the first place, rather a missing feature. It's also not in high demand since the manual scanner works well for the task, and adding it would slow down all -other- scanning.

**Comment by markwkidd** [comment]
Arcade matching is fully implemented in retroarch, and there are databases
for each of the libretro mame cores. A name plus dat matching mode is also
available through the manual scanner. There's no way to better support
arcade scanning, it's very fully supported as is.

There is evidently a bug with the flatpack version. Are you sure you have
downloaded the playlist scanning databases and that they're accessible
locally? Totally sure you not only have MAME ROMs, but also that they're
torrentzipped?

**Comment by Sanaki** [comment]
> There's no way to better support arcade scanning, it's very fully supported as is.

I disagree with this statement, but I also don't believe it's worth implementing better arcade scanning considering the overall effort required for minimal gain and the inevitable requirement that it be locked behind a setting to keep from slowing down the scans of everyone other than the extremely limited target audience.

**Comment by markwkidd** [comment]
Arcade scanning is already implemented. It took a lot of effort, but it's
been done.

**Comment by marcecj** [comment]
I had forgotten to update the cores, databases, etc., but that didn't change anything.  To be clear: scanning for new content does work, but only when I enable "Scan without core match".  I tested this by deleting the "MAME" playlist, which does not get created when "Scan without core match" is off, but then does after I turn it back on again.  I also tried it after installing *all* MAME variants from the core downloader, but that didn't help.

**Comment by markwkidd** [comment]
Could you download Alien Arena from the Arcade section of the retroarch
online updater, then scan that romset? It's known to match the database.

I should have started by asking this.

**Comment by marcecj** [comment]
> Could you download Alien Arena from the Arcade section of the retroarch online updater, then scan that romset? It's known to match the database.

That worked, the game was added to the "Arcade Games" playlist.

**Comment by markwkidd** [comment]
Based on the fact that a known good Alien Arena romset does scan
correctly, I believe your issue is most likely that you do not have romsets
that are built and processed correctly for the versions of MAME that
RetroArch supports. For example, you may have romsets in the 7z file format
as opposed to zip, or they are built for a different version of MAME.

Can you check out these docs?
https://docs.libretro.com/guides/arcade-getting-started/

I am pretty confident that if you start off with a different collection of
romsets than you are using, you will be able to scan them with RetroArch.

**Comment by OENIGMA** [comment]
> Enable "Scan without core match" in Playlists settings and they'll scan. That said, manual scan is still the best route, so long as you feed it a dat.

This Actually works. Thanks man!
